sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/GlobalAddressResolver.java

```diff
@@ -101,23 +102,19 @@ public GlobalAddressResolver(
     }
 
     @Override
-    public void remove(
-        final RxDocumentServiceRequest request,
-        final Set<PartitionKeyRangeIdentity> partitionKeyRangeIdentitySet) {
+    public void updateAddresses(final RxDocumentServiceRequest request, final URI serverKey) {
```

Comment:
don't we need a check inside the method if the feature flag is enabled here?

or if the feature flag is disable this method never is invoked?

Reply:
By theory, this method will never be called when the feature flag is disabled. It is guarded in the following code:
` private void onResponse(Throwable exception, RntbdRequestRecord record) {
        if (exception == null) {
            this.lastSuccessfulRequestNanoTime.set(System.nanoTime());
            return;
        }

        if (this.connectionStateListener != null) {
            this.connectionStateListener.onException(record.args().serviceRequest(), exception);
        }`

But it is good to check here as well, so will change to log an error message in the next iterator.